C#/fgt_sdk_csharp/fgtSdk.cs

```diff
@@ -0,0 +1,1216 @@
+                _fgt_get_pressureStatus = Marshal.GetDelegateForFunctionPointer<fgt_get_pressureStatus>(p_fgt_get_pressureStatus);
+
+                var p_fgt_get_sensorStatus = GetProcAddress(pDll, "fgt_get_sensorStatus");
+                _fgt_get_sensorStatus = Marshal.GetDelegateForFunctionPointer<fgt_get_sensorStatus>(p_fgt_get_sensorStatus);
+
+                var p_fgt_set_power = GetProcAddress(pDll, "fgt_set_power");
+                _fgt_set_power = Marshal.GetDelegateForFunctionPointer<fgt_set_power>(p_fgt_set_power);
+
+                var p_fgt_get_power = GetProcAddress(pDll, "fgt_get_power");
+                _fgt_get_power = Marshal.GetDelegateForFunctionPointer<fgt_get_power>(p_fgt_get_power);
+
+                #endregion
+
+                #region TTL functions
+
+                var p_fgt_set_TtlMode = GetProcAddress(pDll, "fgt_set_TtlMode");
+                _fgt_set_TtlMode = Marshal.GetDelegateForFunctionPointer<fgt_set_TtlMode>(p_fgt_set_TtlMode);
+
+                var p_fgt_read_Ttl = GetProcAddress(pDll, "fgt_read_Ttl");
+                _fgt_read_Ttl = Marshal.GetDelegateForFunctionPointer<fgt_read_Ttl>(p_fgt_read_Ttl);
+
+                var p_fgt_trigger_Ttl = GetProcAddress(pDll, "fgt_trigger_Ttl");
+                _fgt_trigger_Ttl = Marshal.GetDelegateForFunctionPointer<fgt_trigger_Ttl>(p_fgt_trigger_Ttl);
+
+                #endregion
+
+                #region Specific functions
+
+                var p_fgt_set_purge = GetProcAddress(pDll, "fgt_set_purge");
+                _fgt_set_purge = Marshal.GetDelegateForFunctionPointer<fgt_set_purge>(p_fgt_set_purge);
+
+                var p_fgt_set_manual = GetProcAddress(pDll, "fgt_set_manual");
+                _fgt_set_manual = Marshal.GetDelegateForFunctionPointer<fgt_set_manual>(p_fgt_set_manual);
+
+                #endregion
+            }
+            catch (Exception e)
+            {
+                Console.Write(e);
+                throw;
+            }
+
+            #endregion
+        }
+
+        #region Private methods
+
+        /// <summary>
+        /// This method logs to the console detailed information in case an error code has been returned by a low level function
+        /// </summary>
+        /// <param name="errorCode">Error code returned by a low level call</param>
+        /// <param name="type">Type of check to perform</param>
+        /// <param name="index">Pressure or sensor index the low level function used</param>
+        /// <returns>The error code <see cref="fgt_ERROR_CODE"/> that was returned by the low level function</returns>
+        private static fgt_ERROR_CODE ErrCheck(fgt_ERROR_CODE errorCode, fgt_ERRCHECK_TYPE type, uint index = 0)
+        {
+            if (errorCode == fgt_ERROR_CODE.OK) return errorCode;
+
+            var st = new StackTrace();
+            var sf = st.GetFrame(1);
+            var callerName = sf.GetMethod().Name;
```

Comment:
Known issue: in Release mode, this will return "Main" as caller name
Even though it does not prevent the code from running at all, the information loses interest. We need to implement another mechanism